src/branch_and_bound.rs

```diff
@@ -0,0 +1,113 @@
+use bitcoin::blockdata::transaction::TxOut;
+
+use rand::{thread_rng, Rng};
+use std::convert::TryInto;
+
+pub struct BranchAndBound<'a> {
+    pub spending_target: u64,
+    pub available_utxos: Vec<&'a TxOut>,
+    pub addressees_num: u64,
+    pub estimated_fees: u64,
```

Comment:
```suggestion
    pub estimated_feerate: u64,
```

From the calculation of the input cost below, I believe this is a feerate, not a fee.